sys-utils/fallocate.c

```diff
@@ -383,6 +383,11 @@ int main(int argc, char **argv)
 	if (optind != argc)
 		errx(EXIT_FAILURE, _("unexpected number of arguments"));
 
+#ifdef HAVE_POSIX_FALLOCATE
+	if (posix && mode != 0)
+		errx(EXIT_FAILURE, _("with --posix only allocation is supported"));
+#endif
+
```

Comment:
There is already code to check for the options collision; unfortunately, commit 833f9a7aae added the 'x' on the incorrect position. We need:

```diff
diff --git a/sys-utils/fallocate.c b/sys-utils/fallocate.c
index ac7c687f2..ded62f4b0 100644
--- a/sys-utils/fallocate.c
+++ b/sys-utils/fallocate.c
@@ -315,7 +315,7 @@ int main(int argc, char **argv)
        static const ul_excl_t excl[] = {       /* rows and cols in ASCII order */
                { 'c', 'd', 'p', 'z' },
                { 'c', 'n' },
-               { 'x', 'c', 'd', 'i', 'n', 'p', 'z'},
+               { 'c', 'd', 'i', 'n', 'p', 'x', 'z'},
                { 0 }
        };
        int excl_st[ARRAY_SIZE(excl)] = UL_EXCL_STATUS_INIT;
```

Reply:
Oh! Applied and tested